pkg/distributor/distributor_test.go

```diff
@@ -992,7 +1002,16 @@ func TestDistributor_PushQuery(t *testing.T) {
 			assert.Nil(t, err)
 
 			series, err := ds[0].QueryStream(ctx, 0, 10, tc.matchers...)
-			assert.Equal(t, tc.expectedError, err)
+
+			if tc.expectedError == nil {
+				assert.Nil(t, err)
+			} else {
+				assert.EqualError(t, err, tc.expectedError.Error())
+
+				// Assert that downstream GRPC statuses are passed back upstream
+				_, ok := httpgrpc.HTTPResponseFromError(err)
+				assert.True(t, ok)
```

Comment:
```suggestion
				assert.True(t, ok, "expected error to be an httpgrpc error")
```

Same suggestion as before

Reply:
also logging what the actual error was may be useful. #buildingAShedForBikes